gflownet/gflownet.py

```diff
@@ -202,66 +194,127 @@ def parameters(self):
+        self.forward_policy or self.backward_policy, depending on backward. The rest
+        are sampled according to the random policy of the environment
+        (model.random_distribution).
+
+        If a batch is provided (and self.mask_invalid_actions) is True, the masks are
+        retrieved from the batch. Otherwise they are computed from the environments.
 
         Args
         ----
         envs : list of GFlowNetEnv or derived
             A list of instances of the environment
 
-        times : dict
-            Dictionary to store times
+        batch_forward : Batch
+            A batch from which obtain required variables (e.g. masks) to avoid
+            recomputing them.
 
         sampling_method : string
-            - model: uses current forward to obtain the sampling probabilities.
-            - uniform: samples uniformly from the action space.
+            - policy: uses current forward to obtain the sampling probabilities.
+            - random: samples purely from a random policy, that is
+                - random_action_prob = 1.0
+              regardless of the value passed as arguments.
 
-        model : torch model
-            Model to use as policy if sampling_method="policy"
-
-        is_forward : bool
-            True if sampling is forward. False if backward.
+        backward : bool
+            True if sampling is backward. False (forward) by default.
 
         temperature : float
-            Temperature to adjust the logits by logits /= temperature
+            Temperature to adjust the logits by logits /= temperature. If None,
+            self.temperature_logits is used.
+
+        random_action_prob : float
+            Probability of sampling random actions. If None, self.random_action_prob is used.
+
+        no_random : bool
+            If True, the samples will strictly be on-policy, that is
+                - temperature = 1.0
+                - random_action_prob = 0.0
+            regardless of the values passed as arguments.
 
-        mask_invalid_actions : list or None
-            List of invalid action masks for the environments in env. Optional, will be
-            computed if not provided.
+        times : dict
+            Dictionary to store times. Currently not implemented.
+
+        Returns
+        -------
+        actions : list of tuples
+            The sampled actions, one for each environment in envs.
         """
-        # TODO: implement backward sampling from forward policy as in old
-        # backward_sample.
+        # Preliminaries
         if sampling_method == "random":
+            assert (
+                no_random is False
+            ), "sampling_method random and no_random True is ambiguous"
             random_action_prob = 1.0
+            temperature = 1.0
+        elif no_random is True:
+            temperature = 1.0
+            random_action_prob = 0.0
+        else:
+            if temperature is not None:
```

Comment:
```suggestion
            if temperature is None:
```
Shouldn't this be `if temperature is None`?

Reply:
You are right, of course. Have tested these options yet :roll_eyes: